funding/manager.go

```diff
@@ -2728,6 +2769,192 @@ func (f *Manager) waitForTimeout(completeChan *channeldb.OpenChannel,
 	}
 }
 
+// checkFundingInputsSpent checks whether the inputs to the funding transaction
+// are spent by another transaction. This will allow the funding manager to
+// forget the channel.
+func (f *Manager) checkFundingInputsSpent(c *channeldb.OpenChannel) error {
+	numInputs := len(c.FundingTxn.TxIn)
+	errChan := make(chan error, numInputs)
```

Comment:
Wonder what would happen if `defer close(done)` is called and yet another error is sent to this `errChan`. Does it mean we have a leaked channel? I think this might happen if we have two inputs being double spent by the same conflict tx, the second input, which is checked inside a goroutine, will send an error.

Meanwhile could we make its buffer size to be 1 and use this pattern in `registerConfilicTx`?
```
select {
case errChan <- err:
case <- doneChan:
}
```